coderkevin/fresh-data
# Operations supporting multiple resources

This PR completes the support for storing multiple resources from a single API method call. Operations now return a `resources` object which has `resourceName` keys, with object for values that can contain `data` or `error` values within. This allows a single operation to receive data from one or more method calls, and update the resource data state with any mixture of information. Examples of use:

1. Saving a page as an array of IDs and saving each object under its own `resourceName` by ID.
2. Validating data coming back from an API and individually saving `{ data }` for successful ones, and `{ error }` for unsuccessful ones.

To Test:
1. `npm install`, `npm run build`, `npm link`
2. `npm test` and ensure all tests pass.
3. cd into `examples/hello-world`, `npm install`, `npm link fresh-data`, `npm start` and ensure operation.
4. cd into `examples/wp-rest-api`, `npm install`, `npm link fresh-data`, `npm start` and ensure operation.

Update: This PR further trims 53 (real) lines of code and 1380 bytes from the minified bundle! 🎉 

The bundle is now 12,458 bytes.
